src/accessories/CampaignImporterExporter.ttslua

```diff
@@ -0,0 +1,302 @@
+      },
+      CastShadows = true
+    },
+    LuaScript = "",
+    LuaScriptState = "",
+    XmlUI = ""
+}
+
+-- colors and order for following tables
+local COLORS = { "White"; "Orange"; "Green"; "Red"; "Agenda" }
+
+-- counter GUIDS (4x damage and 4x horror)
+local DAMAGE_HORROR_GUIDS = {
+    "eb08d6"; "e64eec"; "1f5a0a"; "591a45";
+    "468e88"; "0257d9"; "7b5729"; "beb964";
+  }
+
+local campaignData = {}
+
+local tokenZone = getObjectFromGUID("ef3b5f")
+local deckImporter = getObjectFromGUID("a28140")
+local investigatorCounter = getObjectFromGUID("f182ee")
+
+local campaignBoxGUID = ""
+local traumaValues = false
+local deckIds = false
+local campaignGuidePage = false
+
+function onLoad(save_state)
+    self.createButton({
+        click_function = "findCampaignFromToken",
+        function_owner = self,
+        label = "Import",
+        position = {x=-1, y=0.2, z=0},
+        width = 350,
+        height = 150,
+        scale = {2, 1, 2},
+    })
+    self.createButton({
+        click_function = "createCampaignToken",
+        function_owner = self,
+        label = "Export",
+        position = {x=1, y=0.2, z=0},
+        width = 350,
+        height = 150,
+        scale = {2, 1, 2},
+    })
+end
+
+function findCampaignFromToken(_, _, _)
+    local coin = nil
+    for _, obj in ipairs(tokenZone.getObjects()) do
+        if obj.hasTag("ImporterToken") then
+            coin = obj
+        end 
+    end
+    if coin == nil then
+        broadcastToAll("Could not find importer token", Color.Red)
+    else
+        local importData = JSON.decode(coin.getGMNotes())
+        campaignBoxGUID = importData[1]
+        local campaignBox = getObjectFromGUID(campaignBoxGUID)
+        if campaignBox.type == "Generic" then
+            campaignBox.call("buttonClick_download")
+        end
+        Wait.condition(
+            function()
+                if #campaignBox.getObjects() > 0 then
+                    placeCampaignFromToken(importData)
+                else
+                    createCampaignFromToken(importData)
+                end
+            end,
+            function()
+                local obj = getObjectFromGUID(campaignBoxGUID)
+                if  obj == nil then 
+                    return false 
+                else
+                    return obj.type == "Bag" and obj.getLuaScript() ~= ""
+                end
+            end,
+            2,
+            function() broadcastToAll("Error loading campaign box") end
+        ) 
+    end
+end
+
+function placeCampaignFromToken(importData)
+    getObjectFromGUID(campaignBoxGUID).call("buttonClick_place")
+    Wait.condition(
+        function() createCampaignFromToken(importData) end,
+        function() return findCampaignLog() ~= nil end,
+        2,
+        function() broadcastToAll("Error placing campaign box") end
+    )
+end
+
+function createCampaignFromToken(importData)
+    findCampaignLog().destruct()
+    findChaosBag().destruct()
```

Comment:
You could instead use the function in `Global` to set the chaos bag state with a string: https://github.com/argonui/SCED/blob/main/src/core/Global.ttslua#L516

This would make the importer/exporter more resilient to future changes to the chaos bag.